### PR TITLE
3 in a row wins

### PR DESCRIPTION
**Pull request by bean00**
*Note: The main interesting behavior I decided on was to return `:X` or `:O` if either player won, or `false` if neither player won (for `which-player-won?`)

- It was easier for me to implement, and made sense to me
- I needed a way to get the player's token if the player won, so I just decided to return the keyword if the player won, and false if the game was either incomplete, or ended it a tie

### CODE REVIEW

**Comment by jdesrosiers** [comment]
Because this PR is dependent on another PR that is still open, there is a risk that someone might merge it before the other one.  Put a note in the description warning your team members not to merge this yet.  Include a link to the dependent PR.  When the other PR is merged and you've rebased this PR, you can remove the note.

File: test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj
Comment (jdesrosiers) [inline test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj]: I like how clean you made these tests.  It's really easy to read.  However, I'm not sure they provide much value.  `get-winning-sets` is just a static data structure.  There is no behavior to test.  Therefore, your tests just end up duplicating the your code.  It would be more valuable to test this through the `has-player-won?` tests.

File: src/clojure_tic_tac_toe/tic_tac_toe_rules.clj
Comment (jdesrosiers) [inline src/clojure_tic_tac_toe/tic_tac_toe_rules.clj]: This doesn't need to be a function.  This is just a static data structure.  There is no reason to call `concat` every time you need this structure.

**Review by jdesrosiers — CHANGES_REQUESTED** [review]

File: test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj
Comment (bean00) [reply test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj]: I decided not to test that `has-player-won?` works for each row, column, and diagonal (I don't think it's necessary). I also removed the `tic_tac_toe_rules_test.clj` file completely, since I don't need it right now.

File: src/clojure_tic_tac_toe/tic_tac_toe_rules.clj
Comment (bean00) [reply src/clojure_tic_tac_toe/tic_tac_toe_rules.clj]: I changed `get-winning-sets` to just be a data structure `winning-sets`

**Comment by bean00** [comment]
@jdesrosiers, I've addressed the changes requested for this PR as well. However, unlike for #4, I didn't combine my commits into just 1 commit by rebasing. For now, I've included the 2 extra commits I made (I figured I can always go back later to rebase them and force push the branch). I decided to do this because I noticed in #4 that GitHub didn't have a clean way of showing that changes were made on the branch.

File: test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj
Comment (tbaik) [reply test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj]: You should probably keep those tests as tests for has-player-won?. Those rules are a pretty important part of your program and it would be good to know if they break, right?
If you deleted some of those winning sets, you won't get any test failures. You definitely would know not to delete some of those winning sets but if this project was touched by more people, they might not know about it and modify those winning sets, which could introduce a bug, provide a bad user experience, and make it a lot harder to find that bug

Comment (bean00) [reply test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj]: Ok, I've tested all the winning sets in `has-player-won?` now.

**Comment by jdesrosiers** [comment]
Something is not quite right with the rebase you did.  The "Add game loop" commit should not be showing up here because it has already been merged to master.  I'm going to approve these changes, but you might want to look into what went wrong before you merge.

**Review by jdesrosiers — APPROVED** [review]

**Comment by bean00** [comment]
Per your comment about the extra "Add game loop" commit, it might have happened because I accidentally committed rather than running `git rebase --continue`. Nevertheless, I checked what changes were introduced, and they were changes that could be combined with the commit right after: `End game when a player gets 3 in a row`. I've rebased those 2 commits together, and will merge this branch into master.

**Pull request merged**
